R/expect_s3_class_linter.R

```diff
@@ -74,6 +87,19 @@ is_s3_class_calls <- paste0("is.", c(
 #'   of S4 objects. [testthat::expect_true()] can also be used for such tests,
 #'   but it is better to use the tailored function instead.
 #'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "expect_true(is(x, 'data.frame'))",
```

Comment:
I wouldn't use `data.frame` here, let's pick something more evocative of S4. Maybe something from Matrix? Or check the methods docs for some common built-in S4 classes

Reply:
`Matrix` should do:

``` r
library(Matrix)
x <- Matrix(0, 3, 2) 
typeof(x)
#> [1] "S4"
is(x, "Matrix")
#> [1] TRUE
```

<sup>Created on 2022-10-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>